provider/pkg/kinds/deprecated.go

```diff
@@ -261,11 +261,6 @@ func ExistsInVersion(gvk *schema.GroupVersionKind, version *cluster.ServerVersio
 func RemovedInVersion(gvk schema.GroupVersionKind) *cluster.ServerVersion {
 	gv, k := groupVersion(gvk.GroupVersion().String()), Kind(gvk.Kind)
 
-	switch k {
-	case PodSecurityPolicy, PodSecurityPolicyList:
-		return &v125
-	}
-
```

Comment:
This code wasn't considering the GroupVersion, and was actually returning the wrong result for `extensions/v1beta1:PodSecurityPolicy` (which was removed in v1.16).